chapters/arrays.tex

```diff
@@ -195,16 +195,17 @@ \section{Built-in Array Functions}\doublelabel{built-in-array-functions}
 \lstinline[mathescape=true]!size(C, $j$)! = $1$ for $n_{\mathrm{A}} + 1 \leq j \leq n$, \lstinline[mathescape=true]!C[$i_{1}$, $\ldots$, $i_{n_{\mathrm{A}}}$, 1, $\ldots$, 1]! =
 \lstinline[mathescape=true]!A[$i_{1}$, $\ldots$, $i_{n_{\mathrm{A}}}$]!\\ \hline
 \end{longtable}
-The argument $n$ must be literal or a constant that can be evaluated during translation because it determines
+The argument $n$ must be a constant that can be evaluated during translation, as it determines
 the number of dimensions of the returned array.
-
 \begin{nonnormative}
-Below, \lstinline!promote! is only used for constant $n$.
-
 Varying $n$ for \lstinline!promote! complicates matrix handling as it
 can change matrix-equations in subtle ways (e.g.\ changing inner products to matrix multiplication).
 \end{nonnormative}
 
+\begin{nonnormative}
+Below, \lstinline!promote! is only used for constant $n$ that can be evaluated during translation.
```

Comment:
But isn't this completely obvious given that it would be illegal to use `promote` otherwise?  I thought the point of this kind of statement would be to mention additional restrictions that don't follow directly from the definition of `promote`.  I get the feeling that the statement should simply be removed.

Reply:
Yeah, I agree it now seems superfluous. There is new text above that explains that n must be constant and even why. So it's not needed to state that the examples follow this rule.

Reply:
Ok, done.